octree/include/pcl/octree/impl/octree_pointcloud.hpp

```diff
@@ -39,11 +39,10 @@
 #ifndef PCL_OCTREE_POINTCLOUD_HPP_
 #define PCL_OCTREE_POINTCLOUD_HPP_
 
-#include <vector>
 #include <assert.h>
 
 #include <pcl/common/common.h>
-
+#include <pcl/octree/impl/octree_base.hpp>
```

Comment:
> But since OctreePointCloudAdjacency, for example, is used somewhere else in PCL with a custom LeafContainer and PCL_NO_PRECOMPILE is turned off, OctreePointCloudAdjacency is not instantiated for these LeafContainers.

Would it not make more sense to make the instantiation there (where the custom leaf container is used) then? The same way it is done in supervoxel clustering. 

This comment of course entailing the change to the include structure in the octree module, to make it uniform with the rest of the library.